pkg/cache/cache.go

```diff
@@ -0,0 +1,26 @@
+// Copyright © 2020 by PACE Telematics GmbH. All rights reserved.
+// Created at 2020/08/12 by Marius Neugebauer
+
+package cache
+
+import (
+	"context"
+	"time"
+)
+
+// Strings caches strings.
+type Strings interface {
+	// SetString stores the value under the key. Any existing value is
+	// overwritten. If forgetAfter is given, the cache automatically forgets the
+	// value after the duration. If forgetAfter is zero then it is never
+	// automatically forgotten.
+	SetString(ctx context.Context, key, value string, forgetAfter time.Duration) error
+
+	// GetString returns the value stored under the key. If there is no value
+	// stored, ErrNotFound is returned.
+	GetString(ctx context.Context, key string) (string, error)
```

Comment:
Wouldn't it be nice to also return how long this item will be valid, i.e., return the ttl/forgetAfter. Because if I fetch an item from the cache that becomes invalid in 5 minutes but I expect my process to run 10 minutes I might want to check for how long the value can be considered valid.

Otherwise I would need to re-fetch the item continuously to check whether it changed or is invalid/gone

Reply:
It would be possible in both the memory and [redis](https://stackoverflow.com/questions/46064445/equivalent-of-setex-for-get-and-ttl) implementation. I'll think about it.

Reply:
That should be a separate func if it involves calling redis twice

Reply:
I implemented the change.